### ISSUE TITLE
Almost all headers are bad placed

### ISSUE THREAD

**Issue by varhub** [JdeRobot/base#256]
Hi all!

Now that we have apt based install that simplifies installation of all libraries next step will be use installed libraries and headers. (towards a sandbox to reduce compilation time (== @jmplaza's TeachingRobotics stripped fork)).

**But this is _impossible_ because a lot of code have incorrect `#include` sentences.**

Examples:
- geometry/math/Line2D.h
  it is placed at include/jderobot, so "correct" #include will be "jderobot/geometry/math/Line2D.h"
  By the way, this also reveals that install process is placing files at invalid places.
  Going back to example, it includes `#include "geoconst.h"`, so to overcome weird includes you must do:
  
  ```
  include_directories(
  /usr/local/include ## the correct one
  /usr/local/include/jderobot ## valid if and only if jderobot acts as a include holder. Then #include <geometry/math/Line2D.h> will work
  /usr/local/include/jderobot/geometry
  /usr/local/include/jderobot/geometry/math
  ...and so on
  )
  ```
  
  This implies a flatten include path that converges in **collisions**. This strategy **must be avoided**.
- parallelIce/cameraClient.h
  It includes this sentence `#include <visionlib/colorspaces/colorspacesmm.h>` that is **only valid** in build time. So nobody can link its development with this library gathered from apt-get.
- Deps/gazebo/CMakeLists.txt
  
  ```
    FOREACH(gazeboInc ${GAZEBO_INCLUDE_DIRS})
            include_directories(${gazeboInc}/gazebo/)
            include_directories(${gazeboInc})
    ENDFOREACH(gazeboInc)
  ```
  
  A complete tricky trick to remove gazebo prefix and so on.
  So instead write `#include <gazebo/physics/X>` you are allowed to write `#include <physics/X>`
- And so on...

This should be revised because is not professional.

**What to do?**
- Decide if all C++ code (interfaces, libraries, tools) must have jderobot prefix
- Decide how and where includes must be placed
  - Check collisions
  - Check context lost (prefix lost and extra prefixes)
- Use correct includes and even refactor headers tree
- Force single include_directories() or even none if it is placed at include's root.
- Eventually, these changes should be complemented with namespacing
  - Should namespaces omit jderobot prefix?
  - _namespacing must not meet exactly file tree if tree is complex_

**Comment by varhub** [comment]
Related to #241 
Related to #257

**Comment by varhub** [comment]
Broken headers of slice interfaces already patched at PR #237
This is why JderobotInterfaces include dir is parched as:
`SET( INTERFACES_CPP_DIR src/stable/interfaces/cpp/jderobot/ src/stable/interfaces/cpp/)`
